node/src/Pos/Block/Logic/Header.hs

```diff
@@ -328,35 +327,89 @@ getHeadersOlderExp upto = do
                 | otherwise = selGo es ii $ succ skipped
         in selGo elems ixs 0
 
--- CSL-396 don't load all the blocks into memory at once
--- | Given @from@ and @to@ headers where @from@ is older (not strict)
--- than @to@, and valid chain in between can be found, headers in
--- range @[from..to]@ will be found.
-getHeadersFromToIncl
-    :: forall ssc m .
-       (HasConfiguration, MonadDBRead m, SscHelpersClass ssc)
-    => HeaderHash -> HeaderHash -> m (Maybe (OldestFirst NE HeaderHash))
-getHeadersFromToIncl older newer = runMaybeT . fmap OldestFirst $ do
+-- | Given optional @depthLimit@, @from@ and @to@ headers where @from@
+-- is older (not strict) than @to@, and valid chain in between can be
+-- found, headers in range @[from..to]@ will be found. If the number
+-- of headers in the chain (which should be returned) is more than
+-- @depthLimit@, error will be thrown.
+getHeadersRange ::
+       forall ssc m. (HasConfiguration, MonadDBRead m, SscHelpersClass ssc)
+    => Maybe Word
+    -> HeaderHash
+    -> HeaderHash
+    -> m (Either Text (OldestFirst NE HeaderHash))
+getHeadersRange _ older newer | older == newer = runExceptT $ do
```

Comment:
I know it's a bit silly, but shouldn't `Just 0` for the depth limit be respected here? The number of headers in the chain is `1 > 0`.

Reply:
It's minor, but you're right. I'll add an extra check.